src/huggingface_hub/file_download.py

```diff
@@ -171,6 +172,41 @@ def get_jinja_version():
     return _jinja_version
 
 
+# Check if the platform supports symlinks
+_are_symlinks_supported = True
+with tempfile.TemporaryDirectory() as tmpdir:
+    src_path = Path(tmpdir) / "dummy_file_src"
+    src_path.touch()
+    dst_path = Path(tmpdir) / "dummy_file_dst"
+    try:
+        os.symlink(src_path, dst_path)
+    except OSError:
```

Comment:
Could this be done at the first call of an on-disk serialization? Otherwise this will be called immediately when importing `huggingface_hub` even if we want to do something entirely different to file download.

Reply:
I thought of doing that on purpose so that the user doesn't have a warning in the middle of a script but rather when starting it. Since it can be disabled anyway, it's not so much of a problem right ? 

But moving it to a function instead is fine as well if you think it makes more sense.

Reply:
I wouldn't find it shocking for it to appear only when calling a serialization method

Reply:
I made the change :) (it's even easier to test :D)